### PR TITLE
[CC-24]: Service Area Selection

### PR DESCRIPTION
**Pull request by jiayuu-u**
### Descriptions

- Link to Jira ticket here: [Jira](https://leonhard.atlassian.net/browse/CC-24)
- Link to Figma: [Figma](https://www.figma.com/file/q27qJBMd20O9HSwJDMwvNi/CourtCanva-UI-Design?node-id=730%3A9483)

### Provide screenshots or videos that the function is working.

https://user-images.githubusercontent.com/68804519/199170475-0445256e-1f0c-415d-9d8c-0fe1ce21139c.mov

### Provide screenshots of no warning, or error in the console.
image

- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket, (i.e `[CC-xx: Feature] Create pull request template`).
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] No unnecessary spaces between lines of code.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Check you named your code properly.
- [x] Remove unused comments.

### CODE REVIEW

File: apps/galaxy/src/services/servicearea/useServiceArea.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+export default function useServiceArea() {
+  const [isLoading, setIsLoading] = useState(false);
+  const [options, setOptions] = useState<Suburb[]>([]);
+  const toast = useToast();
+  const router = useRouter();
+
+  const getSuburbsInfo = async () => {
+    try {
+      const response = await axios.get("/suburbs");
```
Comment (yuanningd) [inline apps/galaxy/src/services/servicearea/useServiceArea.ts:33]: ` const response = await axios.get<Suburb[]>("/suburbs");`

File: apps/galaxy/src/services/servicearea/useServiceArea.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      const result = response.data.suburbs;
```
Comment (yuanningd) [inline apps/galaxy/src/services/servicearea/useServiceArea.ts:34]: try this `const result:  Suburb[]= response.data.suburbs;`

File: apps/galaxy/src/modules/ServiceAreaSelectionForm/ServiceAreaSelectionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { Button, HStack, Container, Select, FormControl, FormLabel } from "@cc/ui-chakra";
+import { useState } from "react";
+import useServiceArea, { FormData } from "../../services/servicearea/useServiceArea";
+import { useForm, Controller } from "react-hook-form";
+import { Select as ReactSelect, createFilter } from "chakra-react-select";
+import { ErrorMessage } from "@hookform/error-message";
+
+const ServiceAreaSelection = () => {
+  const [menuIsOpen, setMenuIsOpen] = useState<boolean>(false);
```
Comment (yuanningd) [inline apps/galaxy/src/modules/ServiceAreaSelectionForm/ServiceAreaSelectionForm.tsx:9]: `<boolean>` is not needed

File: apps/galaxy/src/__tests__/modules/ServiceAreaSelectionForm/ServiceAreaSelection.test.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    renderWithMockedProvider(<ServiceAreaSelection />);
+    expect(screen.getByText("CourtCanva")).toBeInTheDocument();
+    expect(screen.getByText("Please select and confirm your service area")).toBeInTheDocument();
+  });
+
+  it("should show error message", async () => {
+    renderWithMockedProvider(<ServiceAreaSelection />);
+    const submitBtn = screen.getByRole("button", { name: /Submit/i });
+    user.click(submitBtn);
+    await waitFor(() =>
```
Comment (yuanningd) [inline apps/galaxy/src/__tests__/modules/ServiceAreaSelectionForm/ServiceAreaSelection.test.tsx:42]: expect(screen.findByText("Please select at least one option")).toBeVisible()

**Review by yuanningd — APPROVED** [review]

**Comment by Cameron933** [comment]
image

**Review by Cameron933 — APPROVED** [review]

**Pull request merged**